update known issues (#355)
appsignal/appsignal-docs: source/support/known-issues/bugsnag-compatibility.html.md

---

> title: Bugsnag Compatibility

Reviewer: ```suggestion
title: Compatibility issue with other instrumentation gems
```

The filename should also be updated to `gem-instrumentation-compatibility.html`. It's not bugsnag specific.

---

## Affected components

> - AppSignal for Ruby, in combination with Bugsnag 6.12.0 and later

Reviewer: ```suggestion
- AppSignal for Ruby, all versions, in combination with:
  - http_logger gem 0.6.0 and later
  - Bugsnag gem 6.12.0 and later
  - Datadog gem
```

> - AppSignal for Ruby gem versions: `v2.9.17` - `v2.11.0` (see [workarounds](#workaround))

Reviewer 2: ```suggestion
- AppSignal for Ruby, in combination with Bugsnag 6.12.0 and later
```

## Description

We've switched to using `Module.prepend` in our net/http integration in version `2.10.5` of our gem. However, that version broke compatibility with other libraries that use the method aliasing approach, like New Relic and Bugsnag. We then reverted that change in `2.10.7`.

Our rake integration uses the method aliasing approach, while Bugsnag uses `Module.prepend` in theirs. When using both approaches in the same method, you'll end up with a stack level too deep error.

## Symptoms

> - It produces system stack errors when running rake tasks

Reviewer 2: ```suggestion
- Running both AppSignal and Bugsnag 6.12.x or later produces SystemStackErrors when running rake tasks
```

> We've switched to using `Module.prepend` in our net/http integration in version `2.10.5` of our gem. However, that version broke compatibility with other libraries that use the method aliasing approach, like New Relic and Bugsnag. We then reverted that change in `2.10.7`.
> 
> Our rake integration uses the method aliasing approach, while Bugsnag uses `Module.prepend` in theirs. When using both approaches in the same method, you'll end up with a stack level too deep error.
> 
> 
> ## Symptoms
> 
> Running both AppSignal and Bugsnag 6.12.x or later produces SystemStackErrors when running rake tasks.

Reviewer: ```suggestion
Upon executing a rake task or performing a Net::HTTP request the app will raise a "stack level too deep (SystemStackError)" error. There may also be other scenarios this "SystemStackError" occurs, as it can potentially occur with every type of instrumentation set up by the AppSignal gem and other gems.

This error caused by the AppSignal gem and other gems with instrumentation having different methods of instrumentation that are incompatible. These different methods cause aliased methods, set up by instrumentation gems, to be continuously called until Ruby's stack has been exhausted.

We're looking into solutions for this issue in [appsignal-ruby issue #603](https://github.com/appsignal/appsignal-ruby/issues/603). This issue also contains more information about the cause of the issue and potential solutions. Please report issues with other gems if encountered in this issue.
```

## Workaround

> Downgrade the Bugsnag to `6.11.x`, which doesn't use `Module.prepend` yet, or removing it altogether.

Reviewer 2: ```suggestion
Downgrade the Bugsnag library to `6.11.1`, which doesn't use `Module.prepend` yet, or remove it altogether.
```

> Downgrade the Bugsnag library to `6.11.1`, which doesn't use `Module.prepend` yet, or remove it altogether.

Reviewer: ```suggestion
- Downgrade the Bugsnag library to `6.11.1`.
- Downgrade other gems that introduce the same problem in newer versions.
- Remove either of the gems until the issue has been resolved.
```

---
title: Compatibility issue with other instrumentation gems
---

## Affected components

- AppSignal for Ruby, all versions, in combination with:
  - http_logger gem 0.6.0 and later
  - Bugsnag gem 6.12.0 and later
  - Datadog gem

## Description

Upon executing a rake task or performing a Net::HTTP request the app will raise a "stack level too deep (SystemStackError)" error. There may also be other scenarios this "SystemStackError" occurs, as it can potentially occur with every type of instrumentation set up by the AppSignal gem and other gems.

> This error caused by the AppSignal gem and other gems with instrumentation having different methods of instrumentation that are incompatible. These different methods cause aliased methods, set up by instrumentation gems, to be continuously called until Ruby's stack has been exhausted.

Reviewer 2: ```suggestion
This error is caused by the AppSignal gem and other gems with instrumentation having different methods of instrumentation that are incompatible. These different methods cause aliased methods, set up by instrumentation gems, to be continuously called until Ruby's stack has been exhausted.
```

We're looking into solutions for this issue in [appsignal-ruby issue #603](https://github.com/appsignal/appsignal-ruby/issues/603). This issue also contains more information about the cause of the issue and potential solutions. Please report issues with other gems if encountered in this issue.

## Workaround

> - Downgrade the Bugsnag library to `6.11.1`.

Reviewer 2: ```suggestion
- Downgrade the Bugsnag library to `6.11.1`.
- Downgrade the http_logger library to `0.5.1`.
```

- Downgrade other gems that introduce the same problem in newer versions.

> - Remove either of the gems until the issue has been resolved.

Reviewer 2: This item in the known issues list was set up as a separate issue for only the issue with Bugsnag (even though it's caused by the same issue as the problems we're seeing with NR) to keep it clear for customers. I'm fine with having one known-issues item for all items related to #603, but I'd really like this to be a list of versions to downgrade to for all libraries we come across in the future.
